### PR TITLE
Lost properties in property existence enforcing

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Fix error on arrays union that was causing result array missing part of the values.
One of the observable effects was that property constraint enforcer missed part of the properties
and it was possible to create nodes with properties set that do not satisfy all defined constraints.

Add missing tests for node key constraint creation and validation.
Move ImpermanentEnterpriseDatabaseRule to correct package.

### CODE REVIEW

File: community/primitive-collections/src/test/java/org/neo4j/collection/primitive/PrimitiveArraysUnionTest.java [hunk 1]
```diff
@@ -47,7 +47,9 @@
                 lhs( 1, 3, 5 ).rhs( 2, 4, 6 ).expect( 1, 2, 3, 4, 5, 6 ),
                 lhs( 1, 2, 3, 5 ).rhs( 2, 4, 6 ).expect( 1, 2, 3, 4, 5, 6 ),
                 lhs( 2, 3, 4, 7, 8, 9, 12, 16, 19 ).rhs( 4, 6, 9, 11, 12, 15 )
-                        .expect( 2, 3, 4, 6, 7, 8, 9, 11, 12, 15, 16, 19 )
+                        .expect( 2, 3, 4, 6, 7, 8, 9, 11, 12, 15, 16, 19 ),
+                lhs( 10, 13 ).rhs( 13, 18 ).expect( 10, 13, 18 ),
+                lhs( 13, 18 ).rhs( 10, 13 ).expect( 10, 13, 18 )
```
Comment (chrisvest) [inline community/primitive-collections/src/test/java/org/neo4j/collection/primitive/PrimitiveArraysUnionTest.java:63]: Maybe also add a few hundred randomly generated test cases?

Comment (MishaDemianenko) [reply community/primitive-collections/src/test/java/org/neo4j/collection/primitive/PrimitiveArraysUnionTest.java:63]: yep thats a good idea!

Comment (MishaDemianenko) [reply community/primitive-collections/src/test/java/org/neo4j/collection/primitive/PrimitiveArraysUnionTest.java:63]: @chrisvest added a bit of random inputs

**Review by chrisvest — APPROVED** [review]

**Comment by chrisvest** [comment]
The red build is due to flaky tests that have nothing to do with the changes here.

**Pull request merged**